lightningd/lightningd.c

```diff
@@ -251,6 +253,38 @@ static void daemonize_but_keep_dir(void)
 	tal_free(cwd);
 }
 
+static void pidfile_create(const struct lightningd *ld)
+{
+        int pid_fd;
+        char *pid;
+        const tal_t *tmpctx = tal_tmpctx(NULL);
+
+        /* Create PID file */
+        pid_fd = open(ld->pidfile, O_WRONLY|O_CREAT, 0640);
+        if (pid_fd < 0)
+                errx(1, "Failed to open PID file: %s", strerror(errno));
```

Comment:
Actually, `err` does exactly this, so perhaps use `err(1, "Opening pid file %s", ld->pidfile);`?

Similarly for the others...